pkg/network/tracer/cilium_lb.go

```diff
@@ -0,0 +1,270 @@
+	ctTCP, ctUDP       *maps.GenericMap[tupleKey4, ctEntry]
+	backendIDToBackend map[uint32]backend
+	stop               chan struct{}
+}
+
+func newCiliumLoadBalancerConntracker(cfg *config.Config) (netlink.Conntracker, error) {
+	if !cfg.EnableCiliumLBConntracker {
+		return netlink.NewNoOpConntracker(), nil
+	}
+
+	ctTCP, err := ebpf.LoadPinnedMap("/sys/fs/bpf/tc/globals/cilium_ct4_global", &ebpf.LoadPinOptions{
+		ReadOnly: true,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("error loading pinned ct TCP map: %w", err)
+	}
+
+	ctUDP, err := ebpf.LoadPinnedMap("/sys/fs/bpf/tc/globals/cilium_ct_any4_global", &ebpf.LoadPinOptions{
+		ReadOnly: true,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("error loading pinned ct UDP map: %w", err)
+	}
+
+	backends, err := ebpf.LoadPinnedMap("/sys/fs/bpf/tc/globals/cilium_lb4_backends_v3", &ebpf.LoadPinOptions{
+		ReadOnly: true,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("error loading pinned backends map: %w", err)
+	}
+
+	clb := &ciliumLoadBalancerConntracker{
+		backendIDToBackend: make(map[uint32]backend),
+	}
+	if clb.ctTCP, err = maps.Map[tupleKey4, ctEntry](ctTCP); err != nil {
+		return nil, fmt.Errorf("could not make generic map for ct TCP map: %w", err)
+	}
+	if clb.ctUDP, err = maps.Map[tupleKey4, ctEntry](ctUDP); err != nil {
+		return nil, fmt.Errorf("could not make generic map for ct UDP map: %w", err)
+	}
+	if clb.backends, err = maps.Map[backend4KeyV3, backend4ValueV3](backends); err != nil {
+		return nil, fmt.Errorf("could not make generic map for backends map: %w", err)
+	}
+
+	clb.stop = make(chan struct{})
+	go func() {
+		tick := time.NewTicker(10 * time.Second)
+		defer tick.Stop()
+		for {
+			select {
+			case <-tick.C:
+				clb.updateBackends()
+			case <-clb.stop:
+				close(clb.stop)
+				return
+			}
+		}
+	}()
+
+	return clb, nil
+}
+
+func ntohs(n uint16) uint16 {
+	return binary.BigEndian.Uint16([]byte{byte(n), byte(n >> 8)})
+}
+
+func htons(n uint16) uint16 {
+	b := make([]byte, 2)
+	binary.BigEndian.PutUint16(b, n)
+	return *(*uint16)(unsafe.Pointer(&b[0]))
+}
+
+func (clb *ciliumLoadBalancerConntracker) updateBackends() {
+	clb.m.Lock()
+	defer clb.m.Unlock()
+
+	it := clb.backends.Iterate()
+	var k backend4KeyV3
+	var v backend4ValueV3
+	for it.Next(&k, &v) {
+		clb.backendIDToBackend[k.ID] = backend{
+			addr: util.AddressFromNetIP(net.IPv4(v.Address[0], v.Address[1], v.Address[2], v.Address[3])),
+			port: ntohs(v.Port),
+		}
+	}
+
+	if it.Err() != nil {
+		log.Warnf("error iterating lb backends map: %s", it.Err())
+	}
+}
+
+func (clb *ciliumLoadBalancerConntracker) Describe(chan<- *prometheus.Desc) {
+}
+
+// Collect returns the current state of all metrics of the collector
+func (clb *ciliumLoadBalancerConntracker) Collect(chan<- prometheus.Metric) {
+}
+
+func (clb *ciliumLoadBalancerConntracker) GetTranslationForConn(c *network.ConnectionTuple) *network.IPTranslation {
+	// TODO: add ipv6 support
```

Comment:
is there a ticket for this? I'm worried it will get lost

Reply:
Yes: https://datadoghq.atlassian.net/browse/NPM-3694

Reply:
great thanks